### PR TITLE
Add PR template for releases

### PR DESCRIPTION
**Pull request by ivalaginja**
This PR adds a PR template for release branches. It will be automatically applied to PRs from branches starting with the string "release", for example "release_v0.1.01".

It also adds instructions in the docs on how to make a new release.

### CODE REVIEW

**Review by johanmazoyer — DISMISSED** [review]

File: .github/pull_request_templates/release.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+## Description
+
+<!--
+Give this PR a name following "Release vX.X.X" in the title above. You will want to pick an appropriate version
+number depending on what the most recent release was, and on how much this release differs from the last one.
+Provide a brief description of this release, and its circumstances, in this section.
```
[the remark this answers is not in the document]
Comment (ivalaginja) [reply .github/pull_request_templates/release.md:6]: "this release", singular --> "its circumstances"

File: .github/pull_request_templates/release.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+hw ... hardware -> on the THD2 testbed  
+sim ... simulations  
+DM3 ... the in-pupil DM
+
+["converging EFC loop" specs TBD]  
+[current idea: reaching a level of 1e-8 in less than 20 iterations and not diverging after 30 iterations]
+
+Each box can be ticked if the corresponding matrix leads to a converging EFC loop, and on hardware without noticeable
+DH degradations caused by the matrix or loop. FDH is a full dark hole, HDH is a half dark hole. The number of modes
+indicate the number of modes you need to use in the matrix inversion when creating your matrices for the tests.
```
[the remark this answers is not in the document]
Comment (ivalaginja) [reply .github/pull_request_templates/release.md:34]: fixed

File: docs/todo.rst [hunk 3]
```diff
@@ -42,6 +42,40 @@ General guidelines:
...
 
+Releases
+----------
+
+Since Asterix is the baseline simulator for the THD2 testbed, we need to ensure that there is always a version of the
+package available that is capable of producing interaction matrices that successfully create
+dark holes on the testbed. Such versions are marked by releases.
+
+The requirements for a release are that such a matrix can be generated without changing any parameters (just using the
+ones provided in a savef parameter file), and the necessary files are generated without errors. Further, this needs to
```
[the remark this answers is not in the document]
Comment (ivalaginja) [reply docs/todo.rst]: fixed

**Comment by ivalaginja** [comment]
@johanmazoyer this is ready for a final review, I added some of the things I put in #154

**Review by johanmazoyer — APPROVED** [review]

**Pull request merged**